drgn/commands/_builtin/crash/vm.py

```diff
@@ -0,0 +1,233 @@
+    code.add_from_import("drgn", "Object")
+    code.add_from_import("drgn.helpers.linux.mm", "for_each_vma, get_task_rss_info")
+    code.add_from_import("drgn.helpers.linux.sched", "task_cpu")
+    code.add_from_import("drgn.helpers.linux.fs", "inode_path")
+    code.add_from_import("drgn.helpers.common.format", "print_table")
+    code.add_from_import("drgn.helpers.linux.pid", "find_task")
+
+    # Build task list
+    code.append("tasks = [\n")
+    if not args.pids_or_tasks:
+        ctx = crash_get_context(prog, None)
+        code.append(f"    {ctx!r},\n")
+    else:
+        pid_list = []
+        task_list = []
+        invalid_args = []
+        for arg in args.pids_or_tasks:
+            try:
+                typ, val = _pid_or_task(arg)
+                if typ == "pid":
+                    pid_list.append(val)
+                elif typ == "task":
+                    task_list.append(val)
+            except (ValueError, CommandArgumentError):
+                invalid_args.append(arg)
+        for pid in pid_list:
+            code.append(f"    find_task(prog, {pid}),\n")
+        for addr in task_list:
+            code.append(f'    Object(prog, "struct task_struct *", value={addr:#x}),\n')
+
+        for arg in invalid_args:
+            code.append(f"    # vm: invalid task or pid value: {arg}\n")
+    code.append("]\n\n")
+
+    # Add processing logic
+    code.append(
+        """\
+for task in tasks:
+    # Get basic task info
+    pid = int(task.pid)
+    task_addr = task.value_()
+    cpu = task_cpu(task)
+    command = task.comm.string_().decode()
+
+    mm=task.mm
+    if mm:
+        # Get memory statistics
+        mm_addr = mm.value_()
+        pgd = mm.pgd.value_()
+        rss_info = get_task_rss_info(task)
+        rss = rss_info.total
+        total_vm = mm.total_vm.value_() * prog['PAGE_SIZE'].value_()
+        # Get VMA info
+        for vma in for_each_vma(mm):
+            vma_addr = vma.value_()
+            vm_start = vma.vm_start.value_()
+            vm_end = vma.vm_end.value_()
+            vm_flags = vma.vm_flags.value_()
+
+            # Get file path
+            if vma.vm_file:
+                try:
+                    file_path = inode_path(vma.vm_file.f_inode).decode()
+                    if not file_path.startswith("/"):
+                        file_path = "/" + file_path
+                except Exception:
+                    file_path = "<unknown>"
+            else:
+                file_path = "[anon]"
+    else:
+        # No memory mapping
+        mm_addr = 0
+        pgd = 0
+        rss = 0
+        total_vm = 0
+"""
+    )
+    code.print()
+
+
+def _print_task_vm_info(prog: Program, task: Object) -> None:
+    """Print virtual memory info for a single task"""
+    pid = int(task.pid)
+    cpu = task_cpu(task)
+    command = task.comm.string_().decode()
+    task_addr = task.value_()
+
+    print(f'PID: {pid}\tTASK: {task_addr:#x}\tCPU: {cpu}\tCOMMAND: "{command}"')
```

Comment:
I just added `drgn.commands.crash.print_task_header()` that you can use for this.